cmd/cluster/resizenode.go

```diff
@@ -0,0 +1,342 @@
+		Short:             "Resize a node. Requires previous login to the target cluster via ocm backplane login <target_cluster>",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	resizeNodeCmd.Flags().StringVar(&ops.node, "node", "", "The node to resize (e.g. ip-127.0.0.1.eu-west-2.compute.internal)")
+	resizeNodeCmd.Flags().StringVar(&ops.newMachineType, "machine-type", "", "The target AWS machine type to resize to (e.g. m4.xlarge)")
+	resizeNodeCmd.Flags().BoolVarP(&ops.verbose, "verbose", "", false, "Verbose output")
+	ops.k8sclusterresourcefactory.AttachCobraCliFlags(resizeNodeCmd)
+	resizeNodeCmd.MarkFlagRequired("cluster-id")
+	resizeNodeCmd.MarkFlagRequired("node")
+	resizeNodeCmd.MarkFlagRequired("machine-type")
+
+	return resizeNodeCmd
+}
+
+func newResizeNodeOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *resizeNodeOptions {
+	return &resizeNodeOptions{
+		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
+			Flags: flags,
+		},
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *resizeNodeOptions) complete(cmd *cobra.Command, _ []string) error {
+	err := CompleteValidation(&o.k8sclusterresourcefactory, o.IOStreams)
+	if err != nil {
+		return err
+	}
+
+	utils.IsValidClusterKey(o.k8sclusterresourcefactory.ClusterID)
+
+	describedCluster, err := utils.DescribeCluster(o.k8sclusterresourcefactory.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	if strings.ToUpper(describedCluster.CloudProvider().ID()) != "AWS" {
+		return fmt.Errorf("This command is only available for AWS clusters")
+	}
+	/*
+		Ideally we would want additional validation here for:
+		- the machine type exists
+		- the node exists on the cluster
+		- this command isn't used on stage
+
+		As this command is idempotent, it will just fail on a later stage if e.g. the
+		machine type doesn't exist and can be re-run.
+	*/
+
+	return nil
+}
+
+func (o *resizeNodeOptions) initAwsCli() *awsprovider.Client {
+	fmt.Println("Attempting to initialize AWS Client. Switching to hive context to get credentials.")
+
+	targetCluster := o.k8sclusterresourcefactory.ClusterID
+	hiveCluster, err := utils.GetHiveCluster(targetCluster)
+	if err != nil {
+		log.Fatalln("Unable to get hive cluster:", err)
+	}
+
+	utils.SwapOCMContext(hiveCluster)
+	if err != nil {
+		log.Fatalln("Unable to swap OC/kubectl config context:", err)
+	}
+
+	defer func() {
+		err := utils.SwapOCMContext(targetCluster)
+		if err != nil {
+			log.Fatalln(err)
+		}
+	}()
+
+	/*
+		AWS Credentials to the specific cluster are fetched from the hive cluster,
+		therfore we always have to be logged in on the corresponding hive shard initially
+		to initialize the AWS Cli.
+	*/
+	awsClient, err := o.k8sclusterresourcefactory.GetCloudProvider(o.verbose)
```

Comment:
As we can see, we need 4 parameter to create an AWS client:
https://github.com/openshift/osdctl/blob/ed23ebad95488c347ab951d3f6f33a47ea466c26/pkg/k8s/clusterresourcefactory.go#L248
- An access key
- A secret access key
- A session token
- The AWS region to target

Playing with command `ocm backplane cloud credentials`, I can see that this command exactly provides this info.
Looking the code for that command I can see that `backplane-api` service is queried for that:
https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/91a2fa5f29f4506a2b07956ae839262f441b3a21/cmd/ocm-backplane/cloud/credentials.go#L149

As you can see, the only parameter is the cluster id (which is an info you have), verb which is used is the `GetCloudCredentials` verb.
Client however is not the OCM client, it is the backplane client which is not yet used in `osdctl`. However creating that client is not hard. All you have to do is:
1. Retrieve backplane URL using OCM client:
https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/91a2fa5f29f4506a2b07956ae839262f441b3a21/cmd/ocm-backplane/cloud/credentials.go#L127
2. Retrieve the OCM access token:
https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/91a2fa5f29f4506a2b07956ae839262f441b3a21/cmd/ocm-backplane/cloud/credentials.go#L136
3. Then create the backplane client:
https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/91a2fa5f29f4506a2b07956ae839262f441b3a21/cmd/ocm-backplane/cloud/credentials.go#L142
4. And finally make the call to retrieve the AWS credentials with backplane client:
https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/91a2fa5f29f4506a2b07956ae839262f441b3a21/cmd/ocm-backplane/cloud/credentials.go#L149

Remark that all first 3 operations require the `utils` package which can be added with the following import:
`gitlab.cee.redhat.com/service/backplane-cli/pkg/utils`
Once added, run `go mod tidy` to update `go.mod` and `go.sum` files.

If you do not want to add this package, you may want to copy some part of the code from this file locally:
https://gitlab.cee.redhat.com/service/backplane-cli/-/blob/91a2fa5f29f4506a2b07956ae839262f441b3a21/pkg/utils/ocmWrapper.go

Reply:
Forgot to say, but it is normally not required to have `ocm backplane tunnel` up and running to have above code running successfully.

Reply:
It is however required to be OCM logged (i.e. `ocm login` must have been successful) with the proposed change. That's hardly an issue as many commands have this prerequisite:
https://github.com/openshift/osdctl/search?q=CreateConnection

Reply:
As already discussed, breaking the user's oc/kubectl context - even if just momentarily - is also my biggest issue with this PR. 
Unfortunately the ocm-backplane repo, where all of this is already implemented in (as you stated, e.g. used in ocm-backplane cloud console), is in our private gitlab. I wonder what it would take to open source it?

I know that @fahlmant is working on a major refactor of all the outside API we're using in this repository, which also AFAIK includes making it trivial to initialize an aws-cli without a context switch. @fahlmant do you maybe have some input on this PR?

I see the following steps we could take to move forward from here:
- wait for the refactor to happen, rebase this PR with proper aws-cli creation without context swap
- implement the aws-cli creation without context swap in this PR
- keep the context switch for now, fix when the refactor is done

Reply:
As an update to the backplane-api: I was able to quickly implement this PR using backplane-api, this gets rid of the oc context swaps. 
However, the API calls require you to be (as far as I can tell) backplane tunneled. This breaks the whole concept of this tool being open source, as the command would only be usable by tunneling via a closed source tool.

Reply:
Pushed the backplane-api refactor.

Reply:
Lots of the commands in this repo require tunneling, but that's also mostly because that's the only way to access our clusters. If someone had a similar setup and was using our tool for it depending on their networking setup they might not need the tunnel.

While we might have this tool that's open source but also purpose built, that doesn't prevent others from seeing our code and learning from it or borrowing ideas from it, so there's still benefits there.

I don't have any other context around why our other toolset is closed source, but it's probably more due to the sensitive nature of the commands there.